### ISSUE TITLE
Consider switch docker installation and update method from package to repository

### ISSUE THREAD

**Issue by pablomendezroyo** [dappnode/DAppNode#392]
## Context
In DAppNode docker is installed and updated using the [package method](https://docs.docker.com/engine/install/debian/#install-from-a-package). This method entails to hardcode the docker engine, cli and api versions required. It is also a more complicated method that may result in issues.

- Install docker in dappnode through package method: https://github.com/dappnode/DAppNode/blob/master/scripts/dappnode_install_pre.sh
- Update docker in dappnode through package method: https://github.com/dappnode/DNP_DAPPMANAGER/blob/f4ca71831276aac608f61c3e4be4e1c3bc63dba7/packages/dappmanager/hostScripts/docker_engine_update.sh

Since Debian Bullseye become stable, Docker does not have incompatibilities with Debian Bullseye and does not make sense at all to install DAppNode through the package method.

## To discuss
Switch docker installation and update methods from [package method](https://docs.docker.com/engine/install/debian/#install-from-a-package) to [repository method](https://docs.docker.com/engine/install/debian/#install-using-the-repository). 

- Advantages:
   - Code will be much more simple
   - Docker can be easily updated
   - No hardcoded versions
   - Avoid dappnode hosts to be unupdated
 
-Disadvantages:
   - Loose control of docker-debian compatibility? (should be fixed now that docker supports officialy Bullseye)
   - ?

- To take into account:
    - Docker update will still need to be done through dappmanager UI, which in the end uses a docker container to update Docker. This entails the issue of Linux process hierarchy, the update still needs to be done with a detached process.

cc: @eduadiez @tropicar @3alpha @dapplion

**Comment by dapplion** [comment]
Sounds good to me! I always install with the repository in my servers. @3alpha any objections?

**Comment by 3alpha** [comment]
No, I also do it as well from repositories. I think issue here was that docker wasn't readily availabe in the stable branch of Debian repos so this manual stuff had to be done. But since it is in stable and for Debian it really means stable, I wouldn't mind changing this.

@eduadiez Am I right about history of the issue?
